src/main/kotlin/domain/strategy/RandomMineBoardGenerateStrategy.kt

```diff
@@ -0,0 +1,49 @@
+package domain.strategy
+
+import domain.Coordinate
+import domain.Field
+import domain.Fields
+import domain.Height
+import domain.Land
+import domain.Mine
+import domain.MineCnt
+import domain.Width
+
+class RandomMineBoardGenerateStrategy : BoardGenerateStrategy {
+    override fun generate(height: Height, width: Width, mineCnt: MineCnt): Fields {
+        val mineCoordinates = getMineCoordinates(height, width, mineCnt)
+        return generateFields(height, width, mineCoordinates)
+    }
+
+    private fun getMineCoordinates(height: Height, width: Width, mineCnt: MineCnt): List<Coordinate> {
+        return (INITIAL_COORDINATE_SIZE until getMaxCoordinateSize(height, width)).shuffled()
+            .take(mineCnt.value)
+            .map { number -> Coordinate(getCoordinateHeight(number, width), getCoordinateWidth(number, width)) }
+    }
+
+    private fun generateFields(height: Height, width: Width, mineCoordinate: List<Coordinate>): Fields {
+        return Fields(generateFieldMap(height, width, mineCoordinate))
+    }
+
+    private fun generateFieldMap(height: Height, width: Width, mineCoordinate: List<Coordinate>): Map<Coordinate, Field> {
+        val map = mutableMapOf<Coordinate, Field>()
+        repeat(getMaxCoordinateSize(height, width)) { number ->
+            val coordinate = Coordinate(getCoordinateHeight(number, width), getCoordinateWidth(number, width))
+            when (coordinate in mineCoordinate) {
+                true -> map[coordinate] = Mine()
+                false -> map[coordinate] = Land()
+            }
+        }
+        return map
```

Comment:
```suggestion
        return (0..height.value).flatMap { h ->
            (0..width.value).map { w ->
                val coordinate = Coordinate(h, w)
                val field = when (coordinate in mineCoordinate) {
                    true -> Mine()
                    false -> Land()
                }
                coordinate to field
            }
        }.toMap()
```
이런 식으로도 작성할 수 있습니다. height, width에 대한 계산이 없어서 더 읽기 간단하다고 느껴지는데 상현님의 생각은 어떠신가요?